### PR TITLE
adding node based setup and teardown scripts

### PR DESCRIPTION
**Pull request by alvadb**

### CODE REVIEW

File: setup.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+var storageAccountName = "sa" + uuidrg.substring(0,19);
+var location = "westus";
+var errorHasOccurred = false;
+var connectionData = "";
+var wasInArmMode = false;
+
+executeAzureCLI(['config', 'list'], function() {
+	executeAzureCLI(['config', 'mode', 'arm'], function() {
+		executeAzureCLI(['group', 'create', resourceGroup, location], function() {
+			executeAzureCLI(['storage', 'account', 'create', storageAccountName, '-g', resourceGroup, '--type', 'RAGRS', '-l', location], function(code) {
```
Comment (yaxia) [inline setup.js]: We just added the support for encryption@rest and Cool storage om CLI. With the change, '--type' is renamed to '--sku-name' and the '--kind' parameter is added (set to "Storage" in your case).

File: setup.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+						stream.once('open', function(fd) {
+						  stream.write('#{"uuid":"' + uuidrg + '"}\n');
+						  stream.write('#StorageConnectionString = UseDevelopmentStorage=true\n');
+						  stream.write('StorageConnectionString = ' + connectionString + '\n');
+						  stream.end();
+						});
+					}catch(ex) {
+					}
+
+					if(!wasInArmMode) {
```
Comment (yaxia) [inline setup.js]: When (!wasInArmMode) is true, it means CLI is in ASM mode. Why set the mode to ASM again?

File: setup.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		executeAzureCLI(['group', 'create', resourceGroup, location], function() {
+			executeAzureCLI(['storage', 'account', 'create', storageAccountName, '-g', resourceGroup, '--type', 'RAGRS', '-l', location], function(code) {
+				executeAzureCLI(['storage', 'account', 'connectionstring', 'show', '--json', storageAccountName, '-g', resourceGroup], function(code) {
+					try {
+						var connectionString = JSON.parse(connectionData).string;
+						var stream = fs.createWriteStream("resources/config.properties");
+						stream.once('open', function(fd) {
+						  stream.write('#{"uuid":"' + uuidrg + '"}\n');
+						  stream.write('#StorageConnectionString = UseDevelopmentStorage=true\n');
+						  stream.write('StorageConnectionString = ' + connectionString + '\n');
```
Comment (yaxia) [inline setup.js]: when UseDevelopmentStorage is set to true, you don't need to have a other segment in the connection string.

File: setup.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+var uuidrg = uuid.v4().toLowerCase().replace(/-/g, '');
+var resourceGroup = "rg" + uuidrg.substring(0,19);
+var storageAccountName = "sa" + uuidrg.substring(0,19);
+var location = "westus";
+var errorHasOccurred = false;
+var connectionData = "";
+var wasInArmMode = false;
+
+executeAzureCLI(['config', 'list'], function() {
+	executeAzureCLI(['config', 'mode', 'arm'], function() {
```
Comment (yaxia) [inline setup.js]: A general comment on this file, the convention for JavaScript is indent 2 spaces.

File: teardown.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      console.log('unable to teardown');
+    }
+  });
+}
+
+function executeAzureCLI(params, closeCallback, stdOutCallback) {
+  var prc = spawn('azure', params);
+  prc.stderr.on('data', (data) => {
+    errorHasOccurred = true;
+    if(!wasInArmMode && !("config mode asm" + params.join(' '))) {
```
Comment (yaxia) [inline teardown.js]: What is the purpose of "!("config mode asm" + params.join(' '))"? "config mode asm" + params.join(' ')" is a string and it makes the "if" condition always false.

File: setup.js (hunk 1 above)
Comment (alvadb) [inline setup.js]: Will fix in subsequent release.

File: setup.js (hunk 2 above)
Comment (alvadb) [inline setup.js]: If the user was previously in ASM mode then we will reset the mode back to its original state.

File: teardown.js (hunk 5 above)
Comment (alvadb) [inline teardown.js]: Agreed, will fix.

File: setup.js (hunk 4 above)
Comment (alvadb) [inline setup.js]: Will fix in subsequent release.

Comment (emgerner-msft) [inline setup.js]: Are you planning on updating this code soon? Basically, what is the timeline to fix this then?

File: setup.js (hunk 2 above)
Comment (yaxia) [inline setup.js]: Makes sense.

File: setup.js (hunk 4 above)
Comment (alvadb) [inline setup.js]: Plan to update it next week.

File: setup.js (hunk 1 above)
Comment (emgerner-msft) [inline setup.js]: Sounds good!

**Pull request merged**